apps/src/javalab/JavalabCaptchaDialog.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import PropTypes from 'prop-types';
+import javalabMsg from '@cdo/javalab/locale';
+import {
+  appendNewlineToConsoleLog,
+  appendOutputLog,
+} from '@cdo/apps/javalab/redux/consoleRedux.ts';
+import {setIsCaptchaDialogOpen} from '@cdo/apps/javalab/redux/javalabRedux.ts';
+import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
+import ReCaptchaDialog from '@cdo/apps/templates/ReCaptchaDialog';
+
+const BLOG_URL = 'https://support.code.org/hc/en-us/articles/6104078305549';
+
+// exported for tests
+export function UnconnectedJavalabCaptchaDialog({
+  isCaptchaDialogOpen,
+  setIsCaptchaDialogOpen,
+  appendNewlineToConsoleLog,
+  appendOutputLog,
+  recaptchaSiteKey,
+}) {
+  // return promise for tests
+  const onCaptchaSubmit = token => {
+    return fetch('/dashboardapi/v1/users/me/verify_captcha', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({'g-recaptcha-response': token}),
+    }).then(() => {
+      appendOutputLog(javalabMsg.verificationSuccessful());
+      appendNewlineToConsoleLog();
+      setIsCaptchaDialogOpen(false);
```

Comment:
can we auto-run the program here rather than making the user click run again? (We would probably need to make sure we tracked whether they clicked 'run' or 'test' and trigger the correct type again)

Reply:
Yeah I wasn't sure about this without a product spec, seemed a little tricky so went for the simpler version first. Let me take a stab and see if I can whip something up!